### PR TITLE
[5832] eMediplan einlesen

### PR DESCRIPTION
**Pull request by sazgin**

### CODE REVIEW

File: at.medevit.elexis.emediplan.core/src/at/medevit/elexis/emediplan/core/internal/EMediplanServiceImpl.java [hunk 1]
```diff
@@ -222,4 +240,171 @@ protected String getDecodedJsonString(@NonNull String encodedJson){
...
+				} catch (JsonSyntaxException e) {
+					/* ignore */
+				}
+				// because version incompatibility of 16A the MedicalData 'Med' attribute will be removed
+				// MedicalData 'Med' has different types in the version 16A 
+				if (json.getAsJsonObject().get("Patient") != null) {
+					json.getAsJsonObject().get("Patient").getAsJsonObject().remove("Med");
+				}
+				u = g.fromJson(json, Medication.class);
+				logger.warn("json parsed successfully - by removing the 'Med' attribute");
```
Comment (huthomas) [inline at.medevit.elexis.emediplan.core/src/at/medevit/elexis/emediplan/core/internal/EMediplanServiceImpl.java]: Refactor move code block into catch, then return statements are not necessary.

File: at.medevit.elexis.emediplan.core/src/at/medevit/elexis/emediplan/core/internal/EMediplanServiceImpl.java [hunk 2]
```diff
@@ -222,4 +240,171 @@ protected String getDecodedJsonString(@NonNull String encodedJson){
...
+			if (execute.size() > 0) {
+				TimeTool now = new TimeTool();
+				now.add(TimeTool.SECOND, 5);
+				
+				execute = execute.parallelStream().filter(p -> !p.isStopped(now))
+					.collect(Collectors.toList());
+				if (execute.size() > 0) {
+					toAdd.exists = true;
+				}
+			}
```
Comment (huthomas) [inline at.medevit.elexis.emediplan.core/src/at/medevit/elexis/emediplan/core/internal/EMediplanServiceImpl.java]: Could be refactored into own method findExistingPrescription.

File: at.medevit.elexis.emediplan.ui/src/at/medevit/elexis/emediplan/ui/ImportEMediplanDialog.java [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+									medicament.artikelstammItem =
+										(ArtikelstammItem) prescription.getArtikel();
+								} else if (ev.getType() == ElexisEvent.EVENT_DELETE
+									|| ev.getType() == ElexisEvent.EVENT_UPDATE) {
+									if (findPresciptionByMedicament(medicament).size() == 0) {
+										medicament.exists = false;
+									}
+									
+								}
+							}
```
Comment (huthomas) [inline at.medevit.elexis.emediplan.ui/src/at/medevit/elexis/emediplan/ui/ImportEMediplanDialog.java]: Isn't this the same functionality as the findExistingPrescription from the comment above? Can we reuse the code of the findPrescriptionByMedicament method?

Comment (sazgin) [inline at.medevit.elexis.emediplan.ui/src/at/medevit/elexis/emediplan/ui/ImportEMediplanDialog.java]: done

**Pull request merged**